packages/analytics/src/client.ts

```diff
@@ -0,0 +1,36 @@
+import { AnalyticsClientConfig, AnalyticsClientInterface, EventProperties, JsonMap } from './types';
+
+export type AnalyticsClientOptions = Pick<
+  AnalyticsClientConfig,
+  'defaultProperties' | 'defaultTraits'
+>;
+
+export class AnalyticsClient<T extends AnalyticsClientInterface> {
+  constructor(
+    private readonly client: T,
+    private readonly options: AnalyticsClientOptions
+  ) {
+    this.client = client;
+    this.options = options;
+  }
+
+  public async track<K extends keyof EventProperties>(event: K, properties: EventProperties[K]) {
+    return this.client.track(event, { ...properties, ...this.options.defaultProperties });
+  }
+
+  public async untypedTrack(event: string, properties?: JsonMap) {
+    return this.client.track(event, { ...properties, ...this.options.defaultProperties });
+  }
```

Comment:
I wonder if this function could enforce_the_naming_convention? Throwing an error if another format is added?

Reply:
Good call, done and added tests for all the behaviors we expect with the client.